tuxedo-core/src/constraint_checker.rs

```diff
@@ -7,11 +7,111 @@ use sp_std::{fmt::Debug, vec::Vec};
 
 use crate::{dynamic_typing::DynamicallyTypedData, types::Output, Verifier};
 use parity_scale_codec::{Decode, Encode};
-use scale_info::TypeInfo;
 #[cfg(feature = "std")]
 use serde::{Deserialize, Serialize};
 use sp_runtime::transaction_validity::TransactionPriority;
 
+/// A type representing a successful result of checking a transaction's constraints.
+#[derive(Debug, Clone, Default, PartialEq)]
+pub struct ConstraintCheckingSuccess<ValueType> {
+    /// The priority of this transaction that should be reported to the transaction pool.
+    pub priority: TransactionPriority,
+    /// An intermediate value that should be passed to an accumulator that track transient intra-block data.
+    pub accumulator_value: ValueType,
+}
+
+impl<ValueType> ConstraintCheckingSuccess<ValueType> {
+    pub fn transform<NewValueType: From<ValueType>>(
+        self,
+    ) -> ConstraintCheckingSuccess<NewValueType> {
+        ConstraintCheckingSuccess::<NewValueType> {
+            priority: self.priority,
+            accumulator_value: self.accumulator_value.into(),
+        }
+    }
+}
+
+impl<ValueType> From<TransactionPriority> for ConstraintCheckingSuccess<ValueType>
+where
+    ValueType: Default,
+{
+    fn from(priority: TransactionPriority) -> ConstraintCheckingSuccess<ValueType> {
+        ConstraintCheckingSuccess::<ValueType> {
+            priority,
+            accumulator_value: ValueType::default(),
+        }
+    }
+}
+
+impl<ValueType> From<(TransactionPriority, ValueType)> for ConstraintCheckingSuccess<ValueType> {
+    fn from(t: (TransactionPriority, ValueType)) -> ConstraintCheckingSuccess<ValueType> {
+        ConstraintCheckingSuccess::<ValueType> {
+            priority: t.0,
+            accumulator_value: t.1,
+        }
+    }
+}
+
+/// An accumulator allows a Tuxedo piece to do some internal bookkeeping during the course
+/// of a single block. The Bookkeeping must be done through this accumulator-like interface
+/// where each transaction yields some intermediate value that is them folded into the accumulator
+/// via som logic that is encapsulated in the impelementation.
+///
+/// Many Tuxedo pieces will not need accumulators. In such a case, the constraint checker should
+/// simply use () as the accumulator type.
+///
+/// Some typical usecases for an accumulator would be making sure that the block author reward
+/// does not exceed the block's transaction fees. Or making sure that a timestamp transaction occurs
+/// at most once in a single block.
+///
+/// The accumulator is reset automatically at the end of every block. It is not possible to store
+/// data here for use in subsequent blocks.
+pub trait Accumulator {
+    /// The type that is given and also the type of the accumulation result.
+    /// I realize that the most general accumulator swill use two different types for those,
+    /// but let's do that iff we ever need it. I probably will need to so I can do a simple counter.
```

Comment:
```suggestion
    /// but let's do that iff we ever need it. Probably will need to so a simple counter can be implemented.
```

I would avoid comments that reference yourself as I.